pylint/lint/pylinter.py

```diff
@@ -609,6 +621,38 @@ def register_checker(self, checker):
         if not getattr(checker, "enabled", True):
             self.disable(checker.name)
 
+    def enable_fail_on_messages(self):
+        vals = self.config.fail_on
+        if not vals:
+            return
+        all_cats = ["C", "R", "W", "E"]
+
+        fail_on_cats = set()
+        fail_on_msgs = set()
+        for val in vals:
+            # If value is a cateogry, add category, else add message
+            if val in all_cats:
+                fail_on_cats.add(val)
+            else:
+                fail_on_msgs.add(val)
+
+        # For every message in every checker, if cat or msg flagged, enable check
+        for all_checkers in self._checkers.values():
+            for checker in all_checkers:
+                msgs = checker.messages
+                for msg in msgs:
+                    if msg.msgid in fail_on_msgs or msg.symbol in fail_on_msgs:
+                        # message id/symbol matched, enable and flag it
+                        self.enable(msg.msgid)
+                        self.fail_on_symbols.append(msg.symbol)
+                    elif msg.msgid[0] in fail_on_cats:
+                        # message starts with a cateogry value, flag (but do not enable) it
+                        self.fail_on_symbols.append(msg.symbol)
+
+    def any_fail_on_issues(self):
+        issue_symbols = list(self.stats["by_msg"])
```

Comment:
Here if we use a generator instead of casting to a list, we'll get better performance as we'll not have to calculate the whole list of issue_symbols all the time.

Reply:
Good point, will update